src/BloomExe/Book/Book.cs

```diff
@@ -150,9 +150,16 @@ public string TitleBestForUserDisplay
 				
 				if (string.IsNullOrEmpty(display))
 				{
+					//the SIL-LEAD project, SHRP (around 2012-2016) had books that just had an English name, before we changed Bloom
+					//to not show English names. But the order was also critical. So we want those old books to go ahead and use their
+					//English names.
+					var englishTitle = title.GetExactAlternative("en").ToLower();
+					var SHRPMatches = new string[] {"p1", "p2", "p3", "p4", "SHRP"};
+					var couldBeOldStyleUgandaSHRPBook = SHRPMatches.Any(m => englishTitle.Contains(m.ToLower()));
+
 					//if this book is one of the ones we're editing in our collection, it really
 					//needs a title in our main language, it would be confusing to show a title from some other langauge
```

Comment:
I think these two lines of comment should come first...you need to get the big picture before reading about the exception.